api/src/shop/stripe_product_price.py

```diff
@@ -68,34 +76,60 @@ def get_stripe_prices(
 
 def eq_makeradmin_stripe_product(makeradmin_product: Product, stripe_product: stripe.Product) -> bool:
     """Check that the essential parts of the product are the same in both makeradmin and stripe"""
+    if stripe_product.id != makeradmin_product.stripe_product_id:
+        raise ValueError(
+            f"Stripe product id {stripe_product.id} and the stripe id, {makeradmin_product.stripe_product_id}, "
+            + f"associated with makeradmin product id {makeradmin_product.id} does not match"
+        )
     return stripe_product.name == makeradmin_product.name
 
 
 def eq_makeradmin_stripe_price(makeradmin_product: Product, stripe_price: stripe.Price, price_type: PriceType) -> bool:
     """Check that the essential parts of the price are the same in both makeradmin and stripe"""
+    lookup_key_for_product = get_stripe_price_lookup_key(makeradmin_product, price_type)
+    if stripe_price.lookup_key != lookup_key_for_product:
+        raise ValueError(
+            f"Stripe price lookup key {stripe_price.lookup_key} and corresponding key from"
+            + f" makeradmin product {makeradmin_product.id} and PriceType {price_type} does not match"
+        )
+
     recurring = makeradmin_to_stripe_recurring(makeradmin_product, price_type)
-    different = []
 
-    if recurring:
-        if stripe_price.recurring is None:
-            return False
-        different.append(stripe_price.recurring.get("interval") != recurring.interval)
-        different.append(stripe_price.recurring.get("interval_count") != recurring.interval_count)
-    different.append(stripe_price.unit_amount != stripe_amount_from_makeradmin_product(makeradmin_product, recurring))
-    different.append(stripe_price.currency != CURRENCY)
-    different.append(stripe_price.metadata.get("price_type") != price_type.value)
-    return not any(different)
+    base_eq_check = (
+        stripe_price.unit_amount == stripe_amount_from_makeradmin_product(makeradmin_product, recurring)
+        and stripe_price.currency == CURRENCY
+        and stripe_price.metadata.get("price_type") == price_type.value
+    )
+
+    # Check that recurring part of the product and price matches
+    if recurring is not None and stripe_price.recurring is not None:
+        # Both recurring are not None so we have a subscription price,
+        # and hence need to compare the subscription related information
+        return (
+            base_eq_check
+            and stripe_price.recurring.get("interval") == recurring.interval
+            and stripe_price.recurring.get("interval_count") == recurring.interval_count
+        )
+    elif recurring is None and stripe_price.recurring is None:
+        # Both recurring are None so we have a regular product
+        return base_eq_check
+    else:
+        # There is a missmatch between the reccuring and stripe_price.recurring being None
+        return False
```

Comment:
If you are going to remove the differences list, then you can organize the code better.

```suggestion
    if 
        stripe_price.unit_amount != stripe_amount_from_makeradmin_product(makeradmin_product, recurring)
        or stripe_price.currency != CURRENCY
        or stripe_price.metadata.get("price_type") != price_type.value
    ):
        return False

    # Check that either both are recurring, or none are recurring
    if (recurring is not None) != (stripe_price.recurring is not None):
        return False

    if recurring is not None:
        return (stripe_price.recurring.get("interval") == recurring.interval
            and stripe_price.recurring.get("interval_count") == recurring.interval_count
        )

    return True
```